### DISCUSSION TITLE
manipulate the padding and margin of all the elements that implement it

### DISCUSSION THREAD

**Discussion by wnunezc** [twbs/bootstrap#33795] [Ideas]
Is there a way to remove XX% (50%) from the padding and marging all the elements they have scss?

I seek to avoid having to change each component in the scss files and from a single variable to be able to recompress the spaces used for padding and margin.

```
$spacing_pading: 100%; //default
$spacing_margin: 100%; //default
```
output:
```
margin-top: 0;
margin-bottom: 1rem;
```

changed in custom dev:

```
$spacing_pading: 75%; //custom 
$spacing_margin: 75%; //custom 
```
output:
```
margin-top: 0;
margin-bottom: 0.75rem;
```

this will help a lot in the use of:

https://getbootstrap.com/docs/5.0/utilities/spacing/

avoiding the developer exhaustive and abusive use of:

```
.mt-0 {
  margin-top: 0 !important;
}

.ms-1 {
  margin-left: ($spacer * .25) !important;
}

.px-2 {
  padding-left: ($spacer * .5) !important;
  padding-right: ($spacer * .5) !important;
}

.p-3 {
  padding: $spacer !important;
}
```

**Comment by mdo** [comment]
There's no single line to use, but the `$spacer` is a great place to start. Modifying that and our margin/padding Sass maps would get you the majority of the way there in v5.

**Reply by wnunezc** [reply]
do you think this idea could be implemented?

**Reply by mdo** [reply]
Possibly! I don't think it's the most straightforward for generating values for things like buttons and other components, but we do want to keep spacing pretty predictable overall. Very down to at least continue to improve the situation.

**Reply by wnunezc** [reply]
I am interested how can I help?
